src/core_write.cpp

```diff
@@ -166,6 +168,16 @@ void TxToUniv(const CTransaction& tx, const uint256& hashBlock, UniValue& entry)
             o.pushKV("asm", ScriptToAsmStr(txin.scriptSig, true));
             o.pushKV("hex", HexStr(txin.scriptSig.begin(), txin.scriptSig.end()));
             in.pushKV("scriptSig", o);
+            if (!tx.wit.IsNull()) {
+                if (!tx.wit.vtxinwit[i].IsNull()) {
```

Comment:
This check is redundant with the check above, and might be worth removing.

It would probably also be good to add an (i < tx.wit.vtxinwit.size()) check or assertion.